ebean-api/src/main/java/io/ebean/bean/EntityBeanIntercept.java

```diff
@@ -381,7 +381,11 @@ public void setLoaded() {
     this.origValues = null;
     for (int i = 0; i < flags.length; i++) {
       flags[i] &= ~(FLAG_CHANGED_PROP + FLAG_ORIG_VALUE_SET);
+      if (mutableHash != null && mutableHash[i] != null) {
+        mutableHash[i].update(owner._ebean_getField(i));
```

Comment:
when we reset the bean (after a save) we need to update the "mutableHash" with the current value. (needs conversion TO json)